src/SprykerEco/Yves/Computop/Handler/ComputopPaymentHandler.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Yves\Computop\Handler;
+
+use Generated\Shared\Transfer\ComputopCreditCardOrderResponseTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Shared\Computop\ComputopConstants;
+
+class ComputopPaymentHandler
+{
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\ComputopCreditCardOrderResponseTransfer $creditCardOrderResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function addPaymentToQuote(QuoteTransfer $quoteTransfer, ComputopCreditCardOrderResponseTransfer $creditCardOrderResponseTransfer)
+    {
+        if ($quoteTransfer->getPayment() !== null) {
+            $quoteTransfer->getPayment()->getComputopCreditCard()->setCreditCardOrderResponse(
+                $creditCardOrderResponseTransfer
+            );
+
+            $this->setPaymentProviderMethodSelection($quoteTransfer);
+        }
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return void
```

Comment:
This method could return the modified transfer again and then you could do

    $quoteTransfer = $this->setPaymentProviderMethodSelection($quoteTransfer);

This would be the immutable approach.